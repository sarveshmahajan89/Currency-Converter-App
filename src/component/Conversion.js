import React from "react";
import Select from 'react-select';
import $ from 'jquery';
import openSocket from 'socket.io-client';
import Toast from 'light-toast';

const socket = openSocket('http://localhost:8080');

class Conversion extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            amount: 1,
            sourceCurrency: '',
            outputCurrency: '',
            conversionResult: '',
            showHistory: false,
            currencyHistory: [],
            currencyList: [{ value: 'USD', label: 'US Dollar', imageUrl: '/images/currency/usd.svg'  },
                { value: 'EUR', label: 'Euro', imageUrl: '/images/currency/eur.svg' },
                { value: 'JPY', label: 'Japanese Yen', imageUrl: '/images/currency/jpy.svg' },
                { value: 'INR', label: 'Indian Rupee', imageUrl: '/images/currency/inr.svg' },
                { value: 'GBP', label: 'British Pound', imageUrl: '/images/currency/gbp.svg' },
                { value: 'CNY', label: 'Chinese Yuan', imageUrl: '/images/currency/cny.svg' },
                { value: 'AUD', label: 'Australian Dollar', imageUrl: '/images/currency/aud.svg' },
                { value: 'CAD', label: 'Canadian Dollar', imageUrl: '/images/currency/cad.svg' },
                { value: 'CHF', label: 'Swiss Franc', imageUrl: '/images/currency/chf.svg' },
                { value: 'AED', label: 'Emirati Dirham', imageUrl: '/images/currency/aed.svg' },
                ]
        };
        this.customRenderer = this.customRenderer.bind(this);
        this.handleSourceCurrency = this.handleSourceCurrency.bind(this);
        this.handleOutputCurrency = this.handleOutputCurrency.bind(this);
        this.getConversionResult = this.getConversionResult.bind(this);
        this.handleAmoutChange = this.handleAmoutChange.bind(this);
        this.getHistoryDetails = this.getHistoryDetails.bind(this);
        this.swapCurrencies = this.swapCurrencies.bind(this);
    };

    listen() {
        let self = this;
        socket.on('currencyHistory', (data)=>{
            let updateData = data;
            updateData.length>10 && updateData.splice(0,1);
            self.setState({ currencyHistory: data });
        });
    }

    componentDidMount() {
        this.listen();
    }

    handleAmoutChange = e => {
        this.setState({ amount: e.target.value });
    };

    swapCurrencies() {
        if (this.state.sourceCurrency && this.state.outputCurrency) {
            const swapSourceVal = this.state.sourceCurrency;
            const swapOutputVal = this.state.outputCurrency;
            this.setState({
                sourceCurrency: swapOutputVal,
                outputCurrency: swapSourceVal
            });
        } else {
            Toast.fail('Please select both the currencies', 1500);
        }
    }

    componentWillUnmount() {
        socket.close();
    }

    updateCurrencyList(selectedCurrency) {
        return this.state.currencyList.filter(item => item.value !== selectedCurrency);
    }
    handleSourceCurrency = sourceCurrency => {
        this.setState({
            sourceCurrency: sourceCurrency.value
        });
    };

    handleOutputCurrency = outputCurrency => {
        this.setState({
            outputCurrency: outputCurrency.value
        });
    };

    getHistoryDetails() {
        this.setState(prevState => ({
            showHistory: !prevState.showHistory
        }))
    }
    getConversionResult() {
        if (!this.state.sourceCurrency || !this.state.outputCurrency) {
            Toast.fail('Please select both the currencies', 1500);
            return;
        }

        let url = 'https://free.currconv.com/api/v7/convert?q={0}&compact=ultra&apiKey=ae990718a7cddc316c71';
        const conversionString = this.state.sourceCurrency +'_'+this.state.outputCurrency;

        url = url.replace(/\{0\}/, conversionString);

        $.ajax({
            url: url,
            dataType: 'json',
            type: 'GET',

            success: data => {
                const result = this.state.amount * data[conversionString];
                socket.emit('currencyHistory', {amount: this.state.amount, conversion: conversionString.replace('_', ' to '), value: data[conversionString]});
                this.setState({
                    conversionResult: result
                });
            },
            error: (xhr, status, err) => {
                Toast.fail(err.toString(), 1500);
                console.error(this.props.url, status, err.toString()); // eslint-disable-line
            }
        });
    }

    customRenderer(option, index) {
        let labelString = option.label;

        return (
            <div className='select-currency-wrapper'>
                <img className="" size="30" src={option.imageUrl} alt="" />
                <div className='select-name'>
                    <span className='select-currency-code' dangerouslySetInnerHTML={{ __html: labelString }} ></span>
                    <span className='select-currency-name'>{option.value}</span>
                </div>
            </div>
        );
    }

    renderHistory() {
        return this.state.currencyHistory.map((item, index) => {
            return (
                <React.Fragment key={'history-'+index}>
                    <li key={'history-li-'+index} className="history-list">
                        <span>Currency: {item.conversion}</span>
                        <span> , Amount: {item.amount}</span>
                        <span> , Unit Value: {item.value}</span>
                    </li>
                </React.Fragment>
            );
        });
    }
    render() {
        const { amount, currencyList, currencyHistory, conversionResult, sourceCurrency, outputCurrency, showHistory } = this.state;
        return (
            <div>
                <div className="row">
                    <div className='heading'><span className="skills">Please select source and output currencies for conversion</span></div>
                </div>
                <div className="row conversion-container">
                    <div className='container-fields'>
                        <label htmlFor="amount" className="label-input">Amount</label>
                        <input className="input-field" id="amount" aria-label="Amount"
                               name="Amount" type="number" value={amount} onChange={this.handleAmoutChange}/>
                    </div>
                    <div className='container-fields'>
                        <label htmlFor="currency-select-1" className="label-input">Source Currency</label>
                        <Select className="select-field" id="currency-select-1"
                                        value={sourceCurrency} onChange={this.handleSourceCurrency}
                                        options={currencyList}
                                        optionRenderer={this.customRenderer} />
                    </div>
                    <div className='container-fields'>
                        <div>
                            <button aria-label="Swap currencies" title='Swap currencies' className="swap-button-field" type="button">
                                <i className="fa fa-exchange fa-2x" aria-hidden="true" onClick={this.swapCurrencies}></i>
                            </button>
                        </div>
                    </div>
                    <div className='container-fields'>
                        <label htmlFor="currency-select-2" className="label-input">Output Currency</label>
                        <Select className="select-field" id="currency-select-2"
                                        value={outputCurrency} onChange={this.handleOutputCurrency}
                                        options={currencyList}
                                        optionRenderer={this.customRenderer} />
                    </div>
                    <div className='container-fields'>
                        <div>
                            <button aria-label="convert currency" title='convert currency' className={currencyHistory.length === 0 ? 'submit-button-field hidden-history' : 'submit-button-field'}
                                    type="button" onClick={this.getConversionResult}>
                                <i className="fa fa-arrow-circle-right fa-3x" aria-hidden="true"></i>
                            </button>
                        </div>
                    </div>
                    {currencyHistory.length>0 && <div className='container-fields'>
                        <div>
                            <button aria-label="currency history" title='currency history' className="submit-button-field" type="button" onClick={this.getHistoryDetails}>
                                <i className="fa fa-history fa-3x" aria-hidden="true"></i>
                            </button>
                        </div>
                    </div>}
                    {conversionResult && <div className='result-container'>
                        <div className=''>
                            <span>{amount}</span> <span>{sourceCurrency}</span> = <span>{conversionResult}</span>
                            <span> {outputCurrency}</span>
                        </div>
                    </div>}
                    {showHistory && <div className='result-container'>
                        <div>
                            <h3>History</h3>
                            <ul>{
                                this.renderHistory()
                                }
                            </ul>
                        </div>
                    </div>}
                </div>
            </div>
        )
    }
}
export default Conversion ;