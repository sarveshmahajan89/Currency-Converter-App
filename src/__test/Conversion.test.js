import React from 'react';

import Conversion from '../component/Conversion.js'
import {shallow} from "enzyme/build/index";

describe('Conversion', function() {
    const component = shallow(<Conversion/>);
    const input = component.find('#amount');
    it('Carousel renders correctly', () => {

        expect(<Conversion/>).toMatchSnapshot();
    });

    it('should render Conversion correctly with shallow', function () {

        expect(component).toMatchSnapshot();
    });

    it('should render Navigation correctly a elements', function () {

        const text = component.find('span.skills').text();
        expect(text).toEqual('Please select source and output currencies for conversion');
    });

    it('fills the input with a default value', function () {

        expect(input.prop("name")).toBe("Amount");
        expect(input.prop("value")).toBe(1);
    });

    it("updates input value when changed", () => {
        const event = { target: { value: 1} };
        input.simulate("change", event);
        expect(input.prop("value")).toBe(1);
    });

    it('should simulate React Select correctly', function () {

        const select = component.find('#currency-select-1');
        select.simulate('keyDown', { keyCode: 40 });
        select.simulate('keyDown', { keyCode: 13 });

    });
});