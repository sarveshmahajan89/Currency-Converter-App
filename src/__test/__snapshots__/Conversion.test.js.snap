// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversion Carousel renders correctly 1`] = `<Conversion />`;

exports[`Conversion should render Conversion correctly with shallow 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="row"
    >
      <div
        className="heading"
      >
        <span
          className="skills"
        >
          Please select source and output currencies for conversion
        </span>
      </div>
    </div>
    <div
      className="row conversion-container"
    >
      <div
        className="container-fields"
      >
        <label
          className="label-input"
          htmlFor="amount"
        >
          Amount
        </label>
        <input
          aria-label="Amount"
          className="input-field"
          id="amount"
          name="Amount"
          onChange={[Function]}
          type="number"
          value={1}
        />
      </div>
      <div
        className="container-fields"
      >
        <label
          className="label-input"
          htmlFor="currency-select-1"
        >
          Source Currency
        </label>
        <Select
          arrowRenderer={[Function]}
          autosize={true}
          backspaceRemoves={true}
          backspaceToRemoveMessage="Press backspace to remove {label}"
          className="select-field"
          clearAllText="Clear all"
          clearRenderer={[Function]}
          clearValueText="Clear value"
          clearable={true}
          closeOnSelect={true}
          deleteRemoves={true}
          delimiter=","
          disabled={false}
          escapeClearsValue={true}
          filterOptions={[Function]}
          id="currency-select-1"
          ignoreAccents={true}
          ignoreCase={true}
          inputProps={Object {}}
          isLoading={false}
          joinValues={false}
          labelKey="label"
          matchPos="any"
          matchProp="any"
          menuBuffer={0}
          menuRenderer={[Function]}
          multi={false}
          noResultsText="No results found"
          onBlurResetsInput={true}
          onChange={[Function]}
          onCloseResetsInput={true}
          onSelectResetsInput={true}
          openOnClick={true}
          optionComponent={[Function]}
          optionRenderer={[Function]}
          options={
            Array [
              Object {
                "imageUrl": "/images/currency/usd.svg",
                "label": "US Dollar",
                "value": "USD",
              },
              Object {
                "imageUrl": "/images/currency/eur.svg",
                "label": "Euro",
                "value": "EUR",
              },
              Object {
                "imageUrl": "/images/currency/jpy.svg",
                "label": "Japanese Yen",
                "value": "JPY",
              },
              Object {
                "imageUrl": "/images/currency/inr.svg",
                "label": "Indian Rupee",
                "value": "INR",
              },
              Object {
                "imageUrl": "/images/currency/gbp.svg",
                "label": "British Pound",
                "value": "GBP",
              },
              Object {
                "imageUrl": "/images/currency/cny.svg",
                "label": "Chinese Yuan",
                "value": "CNY",
              },
              Object {
                "imageUrl": "/images/currency/aud.svg",
                "label": "Australian Dollar",
                "value": "AUD",
              },
              Object {
                "imageUrl": "/images/currency/cad.svg",
                "label": "Canadian Dollar",
                "value": "CAD",
              },
              Object {
                "imageUrl": "/images/currency/chf.svg",
                "label": "Swiss Franc",
                "value": "CHF",
              },
              Object {
                "imageUrl": "/images/currency/aed.svg",
                "label": "Emirati Dirham",
                "value": "AED",
              },
            ]
          }
          pageSize={5}
          placeholder="Select..."
          removeSelected={true}
          required={false}
          rtl={false}
          scrollMenuIntoView={true}
          searchable={true}
          simpleValue={false}
          tabSelectsValue={true}
          trimFilter={true}
          value=""
          valueComponent={[Function]}
          valueKey="value"
        />
      </div>
      <div
        className="container-fields"
      >
        <div>
          <button
            aria-label="Swap currencies"
            className="swap-button-field"
            title="Swap currencies"
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-exchange fa-2x"
              onClick={[Function]}
            />
          </button>
        </div>
      </div>
      <div
        className="container-fields"
      >
        <label
          className="label-input"
          htmlFor="currency-select-2"
        >
          Output Currency
        </label>
        <Select
          arrowRenderer={[Function]}
          autosize={true}
          backspaceRemoves={true}
          backspaceToRemoveMessage="Press backspace to remove {label}"
          className="select-field"
          clearAllText="Clear all"
          clearRenderer={[Function]}
          clearValueText="Clear value"
          clearable={true}
          closeOnSelect={true}
          deleteRemoves={true}
          delimiter=","
          disabled={false}
          escapeClearsValue={true}
          filterOptions={[Function]}
          id="currency-select-2"
          ignoreAccents={true}
          ignoreCase={true}
          inputProps={Object {}}
          isLoading={false}
          joinValues={false}
          labelKey="label"
          matchPos="any"
          matchProp="any"
          menuBuffer={0}
          menuRenderer={[Function]}
          multi={false}
          noResultsText="No results found"
          onBlurResetsInput={true}
          onChange={[Function]}
          onCloseResetsInput={true}
          onSelectResetsInput={true}
          openOnClick={true}
          optionComponent={[Function]}
          optionRenderer={[Function]}
          options={
            Array [
              Object {
                "imageUrl": "/images/currency/usd.svg",
                "label": "US Dollar",
                "value": "USD",
              },
              Object {
                "imageUrl": "/images/currency/eur.svg",
                "label": "Euro",
                "value": "EUR",
              },
              Object {
                "imageUrl": "/images/currency/jpy.svg",
                "label": "Japanese Yen",
                "value": "JPY",
              },
              Object {
                "imageUrl": "/images/currency/inr.svg",
                "label": "Indian Rupee",
                "value": "INR",
              },
              Object {
                "imageUrl": "/images/currency/gbp.svg",
                "label": "British Pound",
                "value": "GBP",
              },
              Object {
                "imageUrl": "/images/currency/cny.svg",
                "label": "Chinese Yuan",
                "value": "CNY",
              },
              Object {
                "imageUrl": "/images/currency/aud.svg",
                "label": "Australian Dollar",
                "value": "AUD",
              },
              Object {
                "imageUrl": "/images/currency/cad.svg",
                "label": "Canadian Dollar",
                "value": "CAD",
              },
              Object {
                "imageUrl": "/images/currency/chf.svg",
                "label": "Swiss Franc",
                "value": "CHF",
              },
              Object {
                "imageUrl": "/images/currency/aed.svg",
                "label": "Emirati Dirham",
                "value": "AED",
              },
            ]
          }
          pageSize={5}
          placeholder="Select..."
          removeSelected={true}
          required={false}
          rtl={false}
          scrollMenuIntoView={true}
          searchable={true}
          simpleValue={false}
          tabSelectsValue={true}
          trimFilter={true}
          value=""
          valueComponent={[Function]}
          valueKey="value"
        />
      </div>
      <div
        className="container-fields"
      >
        <div>
          <button
            aria-label="convert currency"
            className="submit-button-field hidden-history"
            onClick={[Function]}
            title="convert currency"
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-arrow-circle-right fa-3x"
            />
          </button>
        </div>
      </div>
      
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="row"
      >
        <div
          className="heading"
        >
          <span
            className="skills"
          >
            Please select source and output currencies for conversion
          </span>
        </div>
      </div>
      <div
        className="row conversion-container"
      >
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="amount"
          >
            Amount
          </label>
          <input
            aria-label="Amount"
            className="input-field"
            id="amount"
            name="Amount"
            onChange={[Function]}
            type="number"
            value={1}
          />
        </div>
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="currency-select-1"
          >
            Source Currency
          </label>
          <Select
            arrowRenderer={[Function]}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            className="select-field"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            id="currency-select-1"
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={false}
            labelKey="label"
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            noResultsText="No results found"
            onBlurResetsInput={true}
            onChange={[Function]}
            onCloseResetsInput={true}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            optionRenderer={[Function]}
            options={
              Array [
                Object {
                  "imageUrl": "/images/currency/usd.svg",
                  "label": "US Dollar",
                  "value": "USD",
                },
                Object {
                  "imageUrl": "/images/currency/eur.svg",
                  "label": "Euro",
                  "value": "EUR",
                },
                Object {
                  "imageUrl": "/images/currency/jpy.svg",
                  "label": "Japanese Yen",
                  "value": "JPY",
                },
                Object {
                  "imageUrl": "/images/currency/inr.svg",
                  "label": "Indian Rupee",
                  "value": "INR",
                },
                Object {
                  "imageUrl": "/images/currency/gbp.svg",
                  "label": "British Pound",
                  "value": "GBP",
                },
                Object {
                  "imageUrl": "/images/currency/cny.svg",
                  "label": "Chinese Yuan",
                  "value": "CNY",
                },
                Object {
                  "imageUrl": "/images/currency/aud.svg",
                  "label": "Australian Dollar",
                  "value": "AUD",
                },
                Object {
                  "imageUrl": "/images/currency/cad.svg",
                  "label": "Canadian Dollar",
                  "value": "CAD",
                },
                Object {
                  "imageUrl": "/images/currency/chf.svg",
                  "label": "Swiss Franc",
                  "value": "CHF",
                },
                Object {
                  "imageUrl": "/images/currency/aed.svg",
                  "label": "Emirati Dirham",
                  "value": "AED",
                },
              ]
            }
            pageSize={5}
            placeholder="Select..."
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            value=""
            valueComponent={[Function]}
            valueKey="value"
          />
        </div>
        <div
          className="container-fields"
        >
          <div>
            <button
              aria-label="Swap currencies"
              className="swap-button-field"
              title="Swap currencies"
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-exchange fa-2x"
                onClick={[Function]}
              />
            </button>
          </div>
        </div>
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="currency-select-2"
          >
            Output Currency
          </label>
          <Select
            arrowRenderer={[Function]}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            className="select-field"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            id="currency-select-2"
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={false}
            labelKey="label"
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            noResultsText="No results found"
            onBlurResetsInput={true}
            onChange={[Function]}
            onCloseResetsInput={true}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            optionRenderer={[Function]}
            options={
              Array [
                Object {
                  "imageUrl": "/images/currency/usd.svg",
                  "label": "US Dollar",
                  "value": "USD",
                },
                Object {
                  "imageUrl": "/images/currency/eur.svg",
                  "label": "Euro",
                  "value": "EUR",
                },
                Object {
                  "imageUrl": "/images/currency/jpy.svg",
                  "label": "Japanese Yen",
                  "value": "JPY",
                },
                Object {
                  "imageUrl": "/images/currency/inr.svg",
                  "label": "Indian Rupee",
                  "value": "INR",
                },
                Object {
                  "imageUrl": "/images/currency/gbp.svg",
                  "label": "British Pound",
                  "value": "GBP",
                },
                Object {
                  "imageUrl": "/images/currency/cny.svg",
                  "label": "Chinese Yuan",
                  "value": "CNY",
                },
                Object {
                  "imageUrl": "/images/currency/aud.svg",
                  "label": "Australian Dollar",
                  "value": "AUD",
                },
                Object {
                  "imageUrl": "/images/currency/cad.svg",
                  "label": "Canadian Dollar",
                  "value": "CAD",
                },
                Object {
                  "imageUrl": "/images/currency/chf.svg",
                  "label": "Swiss Franc",
                  "value": "CHF",
                },
                Object {
                  "imageUrl": "/images/currency/aed.svg",
                  "label": "Emirati Dirham",
                  "value": "AED",
                },
              ]
            }
            pageSize={5}
            placeholder="Select..."
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            value=""
            valueComponent={[Function]}
            valueKey="value"
          />
        </div>
        <div
          className="container-fields"
        >
          <div>
            <button
              aria-label="convert currency"
              className="submit-button-field hidden-history"
              onClick={[Function]}
              title="convert currency"
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-arrow-circle-right fa-3x"
              />
            </button>
          </div>
        </div>
        
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <Conversion />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": Conversion {
      "context": Object {},
      "customRenderer": [Function],
      "getConversionResult": [Function],
      "getHistoryDetails": [Function],
      "handleAmoutChange": [Function],
      "handleOutputCurrency": [Function],
      "handleSourceCurrency": [Function],
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "amount": 1,
        "conversionResult": "",
        "currencyHistory": Array [],
        "currencyList": Array [
          Object {
            "imageUrl": "/images/currency/usd.svg",
            "label": "US Dollar",
            "value": "USD",
          },
          Object {
            "imageUrl": "/images/currency/eur.svg",
            "label": "Euro",
            "value": "EUR",
          },
          Object {
            "imageUrl": "/images/currency/jpy.svg",
            "label": "Japanese Yen",
            "value": "JPY",
          },
          Object {
            "imageUrl": "/images/currency/inr.svg",
            "label": "Indian Rupee",
            "value": "INR",
          },
          Object {
            "imageUrl": "/images/currency/gbp.svg",
            "label": "British Pound",
            "value": "GBP",
          },
          Object {
            "imageUrl": "/images/currency/cny.svg",
            "label": "Chinese Yuan",
            "value": "CNY",
          },
          Object {
            "imageUrl": "/images/currency/aud.svg",
            "label": "Australian Dollar",
            "value": "AUD",
          },
          Object {
            "imageUrl": "/images/currency/cad.svg",
            "label": "Canadian Dollar",
            "value": "CAD",
          },
          Object {
            "imageUrl": "/images/currency/chf.svg",
            "label": "Swiss Franc",
            "value": "CHF",
          },
          Object {
            "imageUrl": "/images/currency/aed.svg",
            "label": "Emirati Dirham",
            "value": "AED",
          },
        ],
        "outputCurrency": "",
        "showHistory": false,
        "sourceCurrency": "",
      },
      "swapCurrencies": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div>
      <div
        className="row"
      >
        <div
          className="heading"
        >
          <span
            className="skills"
          >
            Please select source and output currencies for conversion
          </span>
        </div>
      </div>
      <div
        className="row conversion-container"
      >
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="amount"
          >
            Amount
          </label>
          <input
            aria-label="Amount"
            className="input-field"
            id="amount"
            name="Amount"
            onChange={[Function]}
            type="number"
            value={1}
          />
        </div>
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="currency-select-1"
          >
            Source Currency
          </label>
          <Select
            arrowRenderer={[Function]}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            className="select-field"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            id="currency-select-1"
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={false}
            labelKey="label"
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            noResultsText="No results found"
            onBlurResetsInput={true}
            onChange={[Function]}
            onCloseResetsInput={true}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            optionRenderer={[Function]}
            options={
              Array [
                Object {
                  "imageUrl": "/images/currency/usd.svg",
                  "label": "US Dollar",
                  "value": "USD",
                },
                Object {
                  "imageUrl": "/images/currency/eur.svg",
                  "label": "Euro",
                  "value": "EUR",
                },
                Object {
                  "imageUrl": "/images/currency/jpy.svg",
                  "label": "Japanese Yen",
                  "value": "JPY",
                },
                Object {
                  "imageUrl": "/images/currency/inr.svg",
                  "label": "Indian Rupee",
                  "value": "INR",
                },
                Object {
                  "imageUrl": "/images/currency/gbp.svg",
                  "label": "British Pound",
                  "value": "GBP",
                },
                Object {
                  "imageUrl": "/images/currency/cny.svg",
                  "label": "Chinese Yuan",
                  "value": "CNY",
                },
                Object {
                  "imageUrl": "/images/currency/aud.svg",
                  "label": "Australian Dollar",
                  "value": "AUD",
                },
                Object {
                  "imageUrl": "/images/currency/cad.svg",
                  "label": "Canadian Dollar",
                  "value": "CAD",
                },
                Object {
                  "imageUrl": "/images/currency/chf.svg",
                  "label": "Swiss Franc",
                  "value": "CHF",
                },
                Object {
                  "imageUrl": "/images/currency/aed.svg",
                  "label": "Emirati Dirham",
                  "value": "AED",
                },
              ]
            }
            pageSize={5}
            placeholder="Select..."
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            value=""
            valueComponent={[Function]}
            valueKey="value"
          />
        </div>
        <div
          className="container-fields"
        >
          <div>
            <button
              aria-label="Swap currencies"
              className="swap-button-field"
              title="Swap currencies"
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-exchange fa-2x"
                onClick={[Function]}
              />
            </button>
          </div>
        </div>
        <div
          className="container-fields"
        >
          <label
            className="label-input"
            htmlFor="currency-select-2"
          >
            Output Currency
          </label>
          <Select
            arrowRenderer={[Function]}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            className="select-field"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            id="currency-select-2"
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={false}
            labelKey="label"
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            noResultsText="No results found"
            onBlurResetsInput={true}
            onChange={[Function]}
            onCloseResetsInput={true}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            optionRenderer={[Function]}
            options={
              Array [
                Object {
                  "imageUrl": "/images/currency/usd.svg",
                  "label": "US Dollar",
                  "value": "USD",
                },
                Object {
                  "imageUrl": "/images/currency/eur.svg",
                  "label": "Euro",
                  "value": "EUR",
                },
                Object {
                  "imageUrl": "/images/currency/jpy.svg",
                  "label": "Japanese Yen",
                  "value": "JPY",
                },
                Object {
                  "imageUrl": "/images/currency/inr.svg",
                  "label": "Indian Rupee",
                  "value": "INR",
                },
                Object {
                  "imageUrl": "/images/currency/gbp.svg",
                  "label": "British Pound",
                  "value": "GBP",
                },
                Object {
                  "imageUrl": "/images/currency/cny.svg",
                  "label": "Chinese Yuan",
                  "value": "CNY",
                },
                Object {
                  "imageUrl": "/images/currency/aud.svg",
                  "label": "Australian Dollar",
                  "value": "AUD",
                },
                Object {
                  "imageUrl": "/images/currency/cad.svg",
                  "label": "Canadian Dollar",
                  "value": "CAD",
                },
                Object {
                  "imageUrl": "/images/currency/chf.svg",
                  "label": "Swiss Franc",
                  "value": "CHF",
                },
                Object {
                  "imageUrl": "/images/currency/aed.svg",
                  "label": "Emirati Dirham",
                  "value": "AED",
                },
              ]
            }
            pageSize={5}
            placeholder="Select..."
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            value=""
            valueComponent={[Function]}
            valueKey="value"
          />
        </div>
        <div
          className="container-fields"
        >
          <div>
            <button
              aria-label="convert currency"
              className="submit-button-field hidden-history"
              onClick={[Function]}
              title="convert currency"
              type="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-arrow-circle-right fa-3x"
              />
            </button>
          </div>
        </div>
        
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Conversion />,
}
`;
